runtime/compiler/env/SystemSegmentProvider.cpp

```diff
@@ -57,10 +69,22 @@ J9::SystemSegmentProvider::SystemSegmentProvider(size_t defaultSegmentSize, size
       throw;
       }
    _systemBytesAllocated += _systemSegmentSize;
+
+   // log this segment provider
```

Comment:
Can't this run in multiple threads concurrently?

Reply:
Then I'll need a monitor to guard this step. Any example for using a monitor here in openj9?

Reply:
You definitely don't need a lock, I meant that you might need atomic increment. There is also a possibility that the segment provider is created while the compilation queue lock is held, I'll have to check that.

Reply:
I remember a segment provider is created for only one compilation thread, so it will not be shared by others? Besides, this line is original from the segment provider code in OpenJ9.

Reply:
My comment is about getting the compilation sequence number (lines 74-75).